ios/browser/api/sync/brave_sync_worker.cc

```diff
@@ -0,0 +1,430 @@
+    setup_service->PrepareForFirstSyncSetup();
+    setup_service->SetFirstSetupComplete(
+        syncer::SyncFirstSetupCompleteSource::ADVANCED_FLOW_CONFIRM);
+  }
+
+  return true;
+}
+
+const syncer::DeviceInfo* BraveSyncWorker::GetLocalDeviceInfo() {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  auto* device_info_service =
+      DeviceInfoSyncServiceFactory::GetForBrowserState(browser_state_);
+
+  if (!device_info_service) {
+    return nullptr;
+  }
+
+  return device_info_service->GetLocalDeviceInfoProvider()->GetLocalDeviceInfo();
+}
+
+std::vector<std::unique_ptr<syncer::DeviceInfo>>
+BraveSyncWorker::GetDeviceList() {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  auto* device_info_service =
+      DeviceInfoSyncServiceFactory::GetForBrowserState(browser_state_);
+
+  if (!device_info_service) {
+    return std::vector<std::unique_ptr<syncer::DeviceInfo>>();
+  }
+
+  syncer::DeviceInfoTracker* tracker =
+      device_info_service->GetDeviceInfoTracker();
+  return tracker->GetAllDeviceInfo();
+}
+
+std::string BraveSyncWorker::GetOrCreateSyncCode() {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  auto* sync_service = GetSyncService();
+  std::string sync_code;
+  if (sync_service) {
+    sync_code = sync_service->GetOrCreateSyncCode();
+  }
+  return sync_code;
+}
+
+bool BraveSyncWorker::IsValidSyncCode(const std::string& sync_code) {
+  std::vector<uint8_t> seed;
+  if (!brave_sync::crypto::PassphraseToBytes32(sync_code, &seed)) {
+    return false;
+  }
+  return seed.size() == SEED_BYTES_COUNT;
+}
+
+bool BraveSyncWorker::SetSyncCode(const std::string& sync_code) {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  if (sync_code.empty()) {
+    return false;
+  }
+
+  auto* sync_service = GetSyncService();
+  if (!sync_service || !sync_service->SetSyncCode(sync_code)) {
+    return false;
+  }
+  return true;
+}
+
+std::string BraveSyncWorker::GetSyncCodeFromHexSeed(const std::string& hex_code_seed) {
+  DCHECK(!hex_code_seed.empty());
+
+  std::vector<uint8_t> bytes;
+		std::string sync_code_words;
+		if (base::HexStringToBytes(hex_code_seed, &bytes)) {
+				DCHECK_EQ(bytes.size(), SEED_BYTES_COUNT);
+				if (bytes.size(), SEED_BYTES_COUNT) {
+						sync_code_words = brave_sync::crypto::PassphraseFromBytes32(bytes);
+						if (sync_code_words.empty()) {
+								VLOG(1) << __func__ << " PassphraseFromBytes32 failed for " << hex_code_seed;
+						}
+				} else {
+						LOG(ERROR) << "wrong seed bytes " << bytes.size();
+				}
+				
+				DCHECK_NE(sync_code_words, "");
+		} else {
+				VLOG(1) << __func__ << " HexStringToBytes failed for " << hex_code_seed;
+		}
+		return sync_code_words;
+}
+
+bool BraveSyncWorker::IsFirstSetupComplete() {
+		syncer::SyncService* sync_service = GetSyncService();
+		return sync_service &&
+										sync_service->GetUserSettings()->IsFirstSetupComplete();
+}
+
+bool BraveSyncWorker::RemoveDevice(const std::string& device_guid) {
```

Comment:
this is nice and seems be useful when brave-core master will be able to remove other devices in chain :+1: